tests/phpunit/testcases/core/functions.php

```diff
@@ -878,59 +878,85 @@ public function override_initial_version() {
+		$this->assertTrue( false !== bp_get_deprecated_functions_versions(), 'Please check the list of `$deprecated_functions_versions` in `bp_get_deprecated_functions_versions()`. There should be one for each file of the `/src/bp-core/deprecated` directory.' );
+	}
+
+	/**
+	 * @group bp_get_deprecated_functions_versions
 	 * @ticket BP8687
 	 * @ticket BP9075
+	 * @ticket BP9210
 	 */
 	public function test_bp_get_deprecated_functions_versions() {
-		$current_version = bp_get_version();
+		$current_version = (float) bp_get_version();
 		$versions        = bp_get_deprecated_functions_versions();
 		$bp              = buddypress();
 
-		// When current major version is the initial version, we should only load 12.0 deprecated functions file.
-		$this->assertSame( $versions, array( 12.0 ), 'Please check the list of `$deprecated_functions_versions` in `bp_get_deprecated_functions_versions()`. There should be one for each file of the `/src/bp-core/deprecated` directory.' );
-
-		$bp->version = '12.1.1';
-
-		// We should load the 2 lasts deprecated functions files.
-		$this->bp_initial_version = '12.0';
-
 		add_filter( 'pre_option__bp_initial_major_version', array( $this, 'override_initial_version' ), 10, 0 );
 
-		$versions = bp_get_deprecated_functions_versions();
-
-		$this->assertContains( 12.0, $versions, '12.0 deprecated functions should be loaded when 12.1.1 was the first installed version.' );
+		// Fresh install.
+		$this->bp_initial_version = $current_version;
+		$expected_versions        = array();
 
-		$this->bp_initial_version = '9.0';
+		// Include 12.0 until version 16.0.
+		if ( $current_version <= (float) 15 ) {
+			$expected_versions = array( 12.0 );
+		}
 
 		$versions = bp_get_deprecated_functions_versions();
 
-		$this->assertContains( 12.0, $versions, '12.0 deprecated functions should be loaded when 12.0 was updated for a minor release' );
+		$this->assertSame( $versions, $expected_versions, 'Deprecated functions are not loaded for fresh installs.' );
 
-		remove_filter( 'pre_option__bp_initial_major_version', array( $this, 'override_initial_version' ), 10, 0 );
+		// BuddyPress updated from previous versions.
+		$expected_versions = array( $current_version - 1, $current_version ); // The 2 last deprecated function files.
 
-		$bp->version = $current_version;
-
-		// We should load the 2 lasts deprecated functions files.
-		$this->bp_initial_version = '8.0';
-
-		add_filter( 'pre_option__bp_initial_major_version', array( $this, 'override_initial_version' ), 10, 0 );
+		// First BuddyPress installed version is older than 11.0. In this case, the initial version is 0.
+		// See https://buddypress.trac.wordpress.org/changeset/13304
+		$this->bp_initial_version = 0;
 
 		$versions = bp_get_deprecated_functions_versions();
 
-		remove_filter( 'pre_option__bp_initial_major_version', array( $this, 'override_initial_version' ), 10, 0 );
+		// Include 12.0 until version 16.0.
```

Comment:
I think you meant 15 here?
```suggestion
		// Include 12.0 until version 15.0.
```